src/databricks/labs/ucx/mixins/fixtures.py

```diff
@@ -638,6 +639,47 @@ def _scim_values(ids: list[str]) -> list[iam.ComplexValue]:
     return [iam.ComplexValue(value=x) for x in ids]
 
 
+def _wait_group_provisioned(interface: AccountGroupsAPI | GroupsAPI, group: Group) -> None:
+    """Wait for a group to be visible via the supplied group interface.
+
+    Due to consistency issues in the group-management APIs, new groups are not always visible in a consistent manner
+    after being created or modified. This method can be used to mitigate against this by checking that a group:
+
+     - Is visible via the `.get()` interface;
+     - Is visible via the `.list()` interface that enumerates groups.
+
+    Visibility is assumed when 2 calls in a row return the expected results.
```

Comment:
Why is the second time needed?

Reply:
The second time is needed because it's not monotonically consistent: after a single call returns the correct result, another one after that can return the previous (stale) results.

The 2 calls is not a guarantee, but getting 2 in a row that succeed means it's much more likely that subsequent calls will also be consistent. Yes, after 2 there's a chance that a third will be wrong again but it's less likely. This approach is what we use elsewhere (during group migration) and is also the approach used by the terraform provider to mitigate the same problem. (And to be clear, it's mitigation not prevention.)